railties/lib/rails/application/configuration.rb

```diff
@@ -164,6 +164,7 @@ def load_defaults(target_version)
 
           if respond_to?(:active_record)
             active_record.has_many_inversing = true
+            active_record.legacy_connection_handling = false
```

Comment:
I think there should be an entry in [`new_framework_defaults_6_1.rb`](https://github.com/rails/rails/blob/7cb451346618811796efce1f8a2bf576b8e4999c/railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_6_1.rb.tt) for this setting. It's not obvious from the deprecation warnings where `legacy_connection_handling = false` should be added, and it's probably a better experience if the user is [prompted by the upgrade guide to enable it](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#configure-framework-defaults).

It's also not clear from the name alone what `legacy_connection_handling = false` does. What do you think of naming it `granular_connection_handling = true` instead?

Reply:
> It's also not clear from the name alone what legacy_connection_handling = false does

I went with `legacy_connection_handling` because it reads like something you don't want to use anymore, whereas `granular_connection_handling` sounds like a nice feature to have. Additionally we maintained global ActiveRecord::Base behavior so using `granular_connection_handling` as the config option reads like we're changing behavior when 99% of apps don't actually use multi-db handling so they don't need to worry about their connections.

>  think there should be an entry in new_framework_defaults_6_1.rb for this setting.

I'll add it, thanks.

> and it's probably a better experience if the user is prompted by the upgrade guide to enable it.

We'll add that when we write the upgrade guide or if there's already a PR I'll add it there. We don't usually update the upgrade guide in master for each feature added.